src/v/storage/parser.cc

```diff
@@ -95,90 +95,90 @@ static ss::future<result<iobuf>> verify_read_iobuf(
 ss::future<result<stop_parser>> continuous_batch_parser::consume_header() {
     auto f = ss::make_ready_future<result<model::record_batch_header>>(_header);
     if (_header.header_crc == 0 && _header.size_bytes == 0) {
-      f = read_iobuf_exactly(_input, model::packed_record_batch_header_size)
-      .then([this](iobuf b) -> result<iobuf> {
-          if (b.empty()) {
-              // benign outcome. happens at end of file
-              return parser_errc::end_of_stream;
-          }
-          if (
-            b.size_bytes() != model::packed_record_batch_header_size) {
-              stlog.error(
-                "Could not parse header. Expected:{}, but Got:{}. "
-                "consumer:{}",
-                model::packed_record_batch_header_size,
-                b.size_bytes(),
-                *_consumer);
-              return parser_errc::input_stream_not_enough_bytes;
-          }
-          return b;
-      })
-      .then([](result<iobuf> b) {
-          if (!b) {
-              return ss::make_ready_future<
-                result<model::record_batch_header>>(b.error());
-          }
-          return ss::make_ready_future<
-            result<model::record_batch_header>>(
-            header_from_iobuf(std::move(b.value())));
-      });
+        f = read_iobuf_exactly(_input, model::packed_record_batch_header_size)
```

Comment:
i think we should try a coro refactor and see how it does. some of these levels became pretty nested